src/mame/drivers/hp_ipc.cpp

```diff
@@ -744,6 +747,11 @@ void hp_ipc_state::machine_reset()
 	m_spkr->cs_w(1);
 	m_spkr->sk_w(0);
 	m_spkr->di_w(0);
+
+	// Load optional ROMs (if any)
+	for (auto& slot : m_rom_slots) {
+		slot->install_read_handler(m_bankdev->space());
+	}
```

Comment:
You can’t safely do this at reset without either having a way to safely uninstall the handlers from the previous run, or guarding against re-installing on reset.  Are these expansion ROMs limited to particular address ranges that you can uninstall everything from before calling this?

Reply:
Yes, you're right. I copied it over from another HP driver that allowed optional ROMs but I forgot the unmap part.